### PR TITLE
[Driver4] Add modules

### PR DESCRIPTION
**Pull request by yito88**
For supporting Cassandra Java driver 4.x

- Replace import paths
- Modify `DataType` comparison
- Replace `BoundStatement` with `BoundStatementBuilder` in `ValueBinder`

### CODE REVIEW

File: src/main/java/com/scalar/database/storage/cassandra4driver/ResultImpl.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+      }
+      list.add(value);
+    }
+    return Optional.of(new Key(list));
+  }
+
+  private static Value convert(Row row, String name, DataType type)
+      throws UnsupportedTypeException {
+    int typeCode = type.getProtocolCode();
+    if (typeCode == DataTypes.BOOLEAN.getProtocolCode()) {
```
Comment (siyopao) [inline src/main/java/com/scalar/database/storage/cassandra4driver/ResultImpl.java]: Just a thought, but would this look nicer as a `switch` statement?

Comment (yito88) [reply src/main/java/com/scalar/database/storage/cassandra4driver/ResultImpl.java]: Thanks!
I thought about that. But I tried it and got `constant expression required` error.

File: src/main/java/com/scalar/database/storage/cassandra4driver/TableMetadata.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  public TableMetadata(
+      com.datastax.oss.driver.api.core.metadata.schema.TableMetadata tableMetadata) {
+    this.partitionKeyNames =
+        ImmutableSet.copyOf(
+            tableMetadata
+                .getPartitionKey()
+                .stream()
+                .map(ColumnMetadata::getName)
+                .map(CqlIdentifier::toString)
```
[the remark this answers is not in the document]
Comment (yito88) [reply src/main/java/com/scalar/database/storage/cassandra4driver/TableMetadata.java]: Thank you! Fixed.

File: src/main/java/com/scalar/database/storage/cassandra4driver/ValueBinder.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+import com.scalar.database.io.ValueVisitor;
+import java.nio.ByteBuffer;
+import javax.annotation.concurrent.NotThreadSafe;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A visitor class to bind {@code Value}s to a {@link BoundStatementBuilder}
+ *
+ * @author Hiroyuki Yamada
```
[the remark this answers is not in the document]
Comment (yito88) [reply src/main/java/com/scalar/database/storage/cassandra4driver/ValueBinder.java]: Thanks! Added.

**Pull request merged**